app-emulation/docker-credential-helpers/docker-credential-helpers-0.6.4.ebuild

```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+DESCRIPTION="A suite of programs to use native stores to keep Docker credentials safe"
+HOMEPAGE="https://github.com/docker/docker-credential-helpers"
+EGO_PN=github.com/docker/docker-credential-helpers
+
+LICENSE="MIT"
+SLOT="0"
+
+if [[ ${PV} = *9999* ]]; then
+	inherit golang-vcs
+else
+	KEYWORDS="amd64 arm64"
+	EGIT_COMMIT="v${PV}"
+	SRC_URI="https://${EGO_PN}/archive/${EGIT_COMMIT}.tar.gz -> ${P}.tar.gz"
+	inherit golang-vcs-snapshot
+fi
+inherit golang-build
+
+IUSE="gnome-keyring pass"
+REQUIRED_USE="|| ( gnome-keyring pass )"
+RESTRICT="test"
+
+DEPEND="gnome-keyring? ( app-crypt/libsecret )"
+
+RDEPEND="(
+	${DEPEND}
+	pass? ( app-admin/pass )
+)
+"
+
+S="${WORKDIR}/${P}/src/${EGO_PN}"
+
+src_compile() {
+	local -x GOPATH="${WORKDIR}/${P}"
+	use gnome-keyring && emake secretservice
+	use pass && emake pass
+}
+
+src_install() {
+	dobin bin/*
+	dodoc CHANGELOG.md MAINTAINERS README.md
+}
+
+pkg_postinst() {
+	if use gnome-keyring; then
+		elog "For gnome-keyring/kwallet add:\n"
+		elog '		"credStore": "secretservice"'"\n"
+	fi
+	if use pass; then
+		elog "For 'pass' add:\n"
+		elog '		"credStore": "pass"'"\n"
+	fi
+	elog "to your ~/.docker/config.json"
+}
```

Comment:
Do they need to be present on compile stage? I wonder if you can just enable both and have some sort of 
`|| ( dep1 dep2)` condition.

I can't build this package, 
```
>>> Compiling source in /var/tmp/portage/app-emulation/docker-credential-helpers-0.6.4/work/docker-credential-helpers-0.6.4/src/github.com/docker/docker-credential-helpers ...
make -j16 -l10 secretservice 
mkdir -p bin
go build -o bin/docker-credential-secretservice secretservice/cmd/main_linux.go
go: github.com/danieljoos/wincred@v1.1.0: Get "https://proxy.golang.org/github.com/danieljoos/wincred/@v/v1.1.0.mod": dial tcp: lookup proxy.golang.org on 192.168.1.1:53: dial udp 192.168.1.1:53: connect: network is unreachable
make: *** [Makefile:26: secretservice] Error 1
 * ERROR: app-emulation/docker-credential-helpers-0.6.4::gentoo failed (compile phase):
 *   emake failed
```
```
>>> Compiling source in /var/tmp/portage/app-emulation/docker-credential-helpers-0.6.4/work/docker-credential-helpers-0.6.4/src/github.com/docker/docker-credential-helpers ...
make -j16 -l10 pass 
mkdir -p bin
go build -o bin/docker-credential-pass pass/cmd/main.go
go: github.com/danieljoos/wincred@v1.1.0: Get "https://proxy.golang.org/github.com/danieljoos/wincred/@v/v1.1.0.mod": dial tcp: lookup proxy.golang.org on 192.168.1.1:53: dial udp 192.168.1.1:53: connect: network is unreachable
make: *** [Makefile:30: pass] Error 1
 * ERROR: app-emulation/docker-credential-helpers-0.6.4::gentoo failed (compile phase):
 *   emake failed
```
So you need to put some USE conditions to your SRC_URI with this, or pre-download support for both.